docs/reference/modules/admin-bar.md

```diff
@@ -0,0 +1,125 @@
+
+**Example**
+```javascript
+// modules/@apostrophecms/admin-bar/index.js
+module.exports = {
+ options: {
+   groups: [
+     {
+       name: 'media',
+       label: 'Media',
+       items: [
+         '@apostrophecms/image',
+         '@apostrophecms/file',
+         '@apostrophecms/image-tag',
+         '@apostrophecms/file-tag'
+       ]
+     }
+   ]
+ }
+};
+```
+This will result in grouping those four core modules into a single dropdown menu displayed as 'Media' on the menu bar.
+![ApostropheCMS admin bar with open dropdown menu titled 'Media'](/images/group-menu.png)
+
+## Featured methods
+
+The following method belongs to this module and may be useful in project-level code. See the [source code](https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/admin-bar/index.js) for all methods that belong to this module.
+<!-- Some are used within the module and would just create noise here. -->
+
+Because this module has an alias, you can call these from another module from the alias path. For example, `self.apos.adminBar.add()`.
+
+### `add(name, label, permission, options)`
+
+Add an item to the menu bar.
+
+The `name` for the item must be unique within the menu bar to avoid conflicts. When the menu item is clicked, the `name` argument will be emitted on `apos.bus` as the value of an `admin-menu-click` event. If this item controls a specific modal, this will be caught by `TheAposModals` to display the correct modal. So, `name` should be the module name with a `:editor` or `:manager` suffix. For example, `@apostrophecms/global:editor`.
+
+**Example**
+```javascript
+apos.bus.$on('admin-menu-click', async (item) => {
+ // Make sure it is the button we care about, leave others to their own handlers
+ if (item !== 'myCustomWidget:editor') {
+   return;
+ }
+ // Custom code for button action
+});
+```
+The `label` will be the button name displayed on the menu bar.
+
+`permission` is optional and takes an object with `action` and `type` properties. If no permissions are present, anyone can perform the action represented by the button. The `action` property dictates what type of action the button will perform. These include `view`, `view-draft`, `edit`, `publish`, `upload-attachment`, and `delete`. The `type` property matches the name of the module the button is managing. This type must have a registered manager.
+
+`options` can take several properties that control the positioning and display of the new menu item.
+
+|  Property | Type | Description |
+|---|---|---|
+| `after` | String | Takes the name of the button - without `:manager` or `:editor` if present - to output the current button after.  |
```

Comment:
I can't find anything in the source to suggest the "without manager or editor" part is correct?

Reply:
At the start of the method, the `name` key in the item is assigned the value of the passed name argument with the `:xxxx` stripped. The passed name without stripping is assigned to the `action` key.
```
const item = {
          name: name.indexOf(':') === -1 ? name : name.split(':')[0],
          action: name,
          label: label,
          permission: permission,
          options: options || {}
        };
```
The presence of the `after` key is then looked for in the `self.items`. I tested passing a name like `@apostrophecms/file:manager` and it borks. Passing `@apostrophecms/file` works.

Reply:
I guess I wait for resolution of the bug before going forward with the documentation?

Reply:
Nah, you can document the working functionality and leave a ticket behind to address this one oversight.

Reply:
Removed the `after` option until fixed.